pkg/master/master.go

```diff
@@ -165,7 +165,14 @@ type thirdPartyEntry struct {
 type RESTOptionsGetter func(resource unversioned.GroupResource) generic.RESTOptions
 
 type RESTStorageProvider interface {
-	NewRESTStorage(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) (genericapiserver.APIGroupInfo, bool)
+	NewRESTStorage(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) (groupInfo genericapiserver.APIGroupInfo, enabled bool)
+}
+
+// RESTStoragePostStartHookProvider is an optional interface in addition to a RESTStorageProvider that can provide a post start hook
+// for the api server
+type RESTStoragePostStartHookProvider interface {
```

Comment:
there's no need to mash these things together. Just make a PostStartHookProvider type.